### ISSUE TITLE
Raise default timeout to start testing

### ISSUE THREAD

**Issue by yannickowow** [redhat-developer/vscode-extension-tester#245]
Hi
I am currently facing issues when updating my tests. 
My mocha configuration sets a timeout of 20000ms, and I update some of my tests for a longer one.
But this error happens when I try to launch a test session.

```ts
  1) "before all" hook in "{root}":
     Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```
Is it possible to override this value in a setting, or raise this default value ?
Thanks.

**Comment by jrichter1** [comment]
There is currently no option to increase this timeout.
The sole purpose of the root hook is to launch vscode. Normally, if it doesn't start within a few seconds, there is something wrong.

Does this happen on a regular basis, or is it just an isolated incident?

**Comment by yannickowow** [comment]
It happens from time to time, like 1 for 5 tests.
It correctly launch VS Code (15+s), but after that, it closes due to timeout.
When it fails, it is because current run takes a long time to load VS Code file setting and mocha setting.

**Comment by jrichter1** [comment]
All the settings are loaded before the browser launches, not even as part of the root hook. Given the whole startup takes about 2 seconds on an old laptop, I'd say there is something wrong with chromedriver.

There should be a log file `test-resources/test.log` with all the chromedriver events. Maybe it can point to the error.

**Comment by yannickowow** [comment]
Here you can find test.log result:

```ts
[11:45:29 AM][INFO] Populating Preferences file: {
   "alternate_error_pages": {
      "enabled": false
   },
   "autofill": {
      "enabled": false
   },
   "browser": {
      "check_default_browser": false
   },
   "distribution": {
      "import_bookmarks": false,
      "import_history": false,
      "import_search_engine": false,
      "make_chrome_default_for_user": false,
      "skip_first_run_ui": true
   },
   "dns_prefetching": {
      "enabled": false
   },
   "profile": {
      "content_settings": {
         "pattern_pairs": {
            "https://*,*": {
               "media-stream": {
                  "audio": "Default",
                  "video": "Default"
               }
            }
         }
      },
      "default_content_setting_values": {
         "geolocation": 1
      },
      "default_content_settings": {
         "geolocation": 1,
         "mouselock": 1,
         "notifications": 1,
         "popups": 1,
         "ppapi-broker": 1
      },
      "password_manager_enabled": false
   },
   "safebrowsing": {
      "enabled": false
   },
   "search": {
      "suggest_enabled": false
   },
   "translate": {
      "enabled": false
   }
}
[11:45:29 AM][INFO] Populating Local State file: {
   "background_mode": {
      "enabled": false
   },
   "ssl": {
      "rev_checking": {
         "enabled": false
      }
   }
}
[11:45:29 AM][INFO] Launching chrome: </path/to/extension>/ui-test-suite/test-resources/VSCode-linux-x64/code --disable-background-networking --disable-client-side-phishing-detection --disable-default-apps --disable-dev-shm-usage --disable-hang-monitor --disable-popup-blocking --disable-prompt-on-repost --disable-sync --enable-automation --enable-blink-features=ShadowDOMV0 --enable-logging --log-level=0 --no-first-run --no-sandbox --no-service-autorun --password-store=basic --remote-debugging-port=0 --test-type=webdriver --use-mock-keychain --user-data-dir=</path/to/extension>/ui-test-suite/test-resources/settings
[11:45:30 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] RESPONSE InitSession {
   "acceptInsecureCerts": false,
   "acceptSslCerts": false,
   "applicationCacheEnabled": false,
   "browserConnectionEnabled": false,
   "browserName": "chrome",
   "chrome": {
      "chromedriverVersion": "87.0.4280.88 (89e2380a3e36c3464b5dd1302349b1382549290d-refs/branch-heads/4280@{#1761})",
      "userDataDir": "</path/to/extension>/ui-test-suite/test-resources/settings"
   },
   "cssSelectorsEnabled": true,
   "databaseEnabled": false,
   "goog:chromeOptions": {
      "debuggerAddress": "localhost:44318"
   },
   "handlesAlerts": true,
   "hasTouchScreen": false,
   "javascriptEnabled": true,
   "locationContextEnabled": true,
   "mobileEmulationEnabled": false,
   "nativeEvents": true,
   "networkConnectionEnabled": false,
   "pageLoadStrategy": "normal",
   "platform": "Linux",
   "proxy": {

   },
   "~~~": "..."
}
[11:45:31 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] COMMAND FindElements {
   "using": "css selector",
   "value": ".monaco-workbench"
}
[11:45:31 AM][INFO] Waiting for pending navigations...
[11:45:31 AM][INFO] Done waiting for pending navigations. Status: ok
[11:45:31 AM][INFO] Waiting for pending navigations...
[11:45:31 AM][INFO] Done waiting for pending navigations. Status: ok
[11:45:31 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] RESPONSE FindElements [  ]
[11:45:31 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] COMMAND FindElements {
   "using": "css selector",
   "value": ".monaco-workbench"
}
[11:45:31 AM][INFO] Waiting for pending navigations...
[11:45:31 AM][INFO] Done waiting for pending navigations. Status: ok
[11:45:32 AM][INFO] Waiting for pending navigations...
[11:45:32 AM][INFO] Done waiting for pending navigations. Status: ok
[11:45:32 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] RESPONSE FindElements [ {
   "ELEMENT": "0.7769907691179099-1"
} ]
[11:45:34 AM][INFO] [bd4be8523db8dec53580f4b3ce4ff306] COMMAND GetLog {
   "type": "driver"
}
```

**Comment by jrichter1** [comment]
Raised the timeout and added some info about the browser starting in the log. Available in `4.0.1`.
